crates/node/src/main.rs

```diff
@@ -50,16 +51,13 @@ async fn main() -> anyhow::Result<()> {
         run_cmd.chain_config()?
     };
 
-    // Check if the devnet is running with the correct chain id.
-    if run_cmd.devnet && chain_config.chain_id != NetworkType::Devnet.chain_id() {
-        if !run_cmd.block_production_params.override_devnet_chain_id {
-            tracing::error!("You're running a devnet with the network config of {:?}. This means that devnet transactions can be replayed on the actual network. Use `--network=devnet` instead. Or if this is the expected behavior please pass `--override-devnet-chain-id`", chain_config.chain_name);
-            panic!();
-        } else {
-            // This log is immediately flooded with devnet accounts and so this can be missed.
-            // Should we add a delay here to make this clearly visisble?
-            tracing::warn!("You're running a devnet with the network config of {:?}. This means that devnet transactions can be replayed on the actual network.", run_cmd.network);
-        }
+    // Check if the devnet is running with the correct chain id. This is purely
+    // to avoid accidental setups which would allow for replay attacks. This is
+    // possible if the devnet has the same chain id as another popular chain,
+    // allowing txs which occur on it to also be replayed on that other chain.
+    if run_cmd.devnet && (chain_config.chain_id == ChainId::Mainnet || chain_config.chain_id == ChainId::Sepolia) {
```

Comment:
do we not want to give an option to force this? like if I really need to run a devnet with sepolia mode just so I can test some script or something else that I have?

Reply:
Idk, this seems liek something which would eventually make its way to prod :sweat_smile:. Any situation in particular where you think this could be useful?